optimize/util/optimize-commons/src/main/java/io/camunda/optimize/util/tomcat/LoggingConfigurationReader.java

```diff
@@ -7,42 +7,46 @@
  */
 package io.camunda.optimize.util.tomcat;
 
-import ch.qos.logback.classic.Level;
-import ch.qos.logback.classic.LoggerContext;
-import ch.qos.logback.classic.joran.JoranConfigurator;
-import ch.qos.logback.core.joran.spi.JoranException;
 import com.google.common.collect.Lists;
+import io.camunda.optimize.service.exceptions.OptimizeRuntimeException;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Arrays;
 import java.util.LinkedList;
 import java.util.Objects;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.Configurator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class LoggingConfigurationReader {
 
+  /* The order of this list is important */
   private final LinkedList<String> loggingConfigNames =
-      Lists.newLinkedList(
-          Arrays.asList("environment-logback.xml", "logback-test.xml", "logback.xml"));
+      Lists.newLinkedList(Arrays.asList("environment-log4j2.xml", "log4j2-test.xml", "log4j2.xml"));
```

Comment:
Also, when starting via `optimize-startup.sh` it logs

```
Starting Camunda Optimize 8.7.0-SNAPSHOT...

SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
2024-12-09 13:32:49.006  INFO 93185 --- [kground-preinit] o.h.v.i.u.Version                        : HV000001: Hibernate Validator 8.0.1.Final
2024-12-09 13:32:49.029  INFO 93185 --- [           main] i.c.o.Main                               : Starting Main v8.7.0-SNAPSHOT using Java 21.0.4 with PID 93185 (/Users/roman.smirnov/Development/projects/camunda/optimize-distro/target/camunda-optimize-8.7.0-SNAPSHOT-production/optimize-backend-8.7.0-SNAPSHOT.jar started by roman.smirnov in /Users/roman.smirnov/Development/projects/camunda/optimize-distro/target/camunda-optimize-8.7.0-SNAPSHOT-production)
2024-12-09 13:32:49.030  INFO 93185 --- [           main] i.c.o.Main                               : The following 1 profile is active: "ccsm"
2024-12-09 13:32:49.809  INFO 93185 --- [           main] o.s.b.w.e.t.TomcatWebServer              : Tomcat initialized with ports 8090 (http), 8091 (https)
Dez. 09, 2024 1:32:49 PM org.apache.coyote.AbstractProtocol init
INFORMATION: Initializing ProtocolHandler ["http-nio-8090"]
Dez. 09, 2024 1:32:49 PM org.apache.coyote.AbstractProtocol init
```

Reply:
About that `environment-log4j2.xml`, that is a list of file names that the customer can define and they are loaded in order. For the moment I simply translated into log4j2 the old behavior optimize had with the old logging system